examples/experimental/dagster-airlift/dagster_airlift/in_airflow/mark_as_migrating.py

```diff
@@ -0,0 +1,105 @@
+import json
+import logging
+from typing import Any, Dict, Optional
+
+from airflow import DAG
+from airflow.models import BaseOperator
+
+from dagster_airlift.in_airflow.dagster_operator import build_dagster_task
+
+from ..migration_state import AirflowMigrationState
+
+
+def mark_as_dagster_migrating(
+    *,
+    global_vars: Dict[str, Any],
+    migration_state: AirflowMigrationState,
+    logger: Optional[logging.Logger] = None,
+) -> None:
+    """Alters all airflow dags in the current context to be marked as migrating to dagster.
+    Uses a migration dictionary to determine the status of the migration for each task within each dag.
+    Should only ever be the last line in a dag file.
+
+    Args:
+        global_vars (Dict[str, Any]): The global variables in the current context. In most cases, retrieved with `globals()` (no import required).
+            This is equivalent to what airflow already does to introspect the dags which exist in a given module context:
+            https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#loading-dags
+        migration_state (AirflowMigrationState): The migration state for the dags.
+        logger (Optional[logging.Logger]): The logger to use. Defaults to logging.getLogger("dagster_airlift").
+    """
+    caller_module = global_vars.get("__module__")
+    suffix = f" in module `{caller_module}`" if caller_module else ""
+    if not logger:
+        logger = logging.getLogger("dagster_airlift")
+    logger.debug(f"Searching for dags migrating to dagster{suffix}...")
+    migrating_dags = set()
+    # Do a pass to collect dags and ensure that migration information is set correctly.
+    for var, obj in global_vars.items():
```

Comment:
similar to previous I think should should move all accesses to `global_vars` to very top and immediately just get all the dags out of these and then write the rest of code in terms of dags.